packages/dd-trace/src/config.js

```diff
@@ -501,6 +501,10 @@ class Config {
     this._setValue(defaults, 'isGitUploadEnabled', false)
     this._setValue(defaults, 'isIntelligentTestRunnerEnabled', false)
     this._setValue(defaults, 'isManualApiEnabled', false)
+    this._setValue(defaults, 'llmobs.agentlessEnabled', false)
+    this._setValue(defaults, 'llmobs.apiKey', undefined)
```

Comment:
I think it's because CI Visibility needs it defined as an environment variable because it changes its own exporter before even initializing the library. That is not necessarily an argument against a programmatic config though, as one doesn't prevent the other. For configuration options that apply library-wide, I'd definitely prefer that it's at the root, otherwise it gives the impression that you can mix and match API keys or even a hybrid agent+agentless.

Reply:
good point. happy to throw that change in this PR (have it at the root level), unless you'd rather have it in it's own PR

Reply:
actually, I can push up a change that has LLMObs only rely on the current `config.apiKey` set from `DD_API_KEY`. I'll propose a follow-up PR to add it to the root level of the in-code config as well

Reply:
addressed in 3e6945b57c39200bbbeb2d9331b9f6dfe20165bf (+ test fix after lol)